setuptools/ccompiler.py

```diff
@@ -0,0 +1,6 @@
+# Expose a subset of distutils as public API
+# to help with migration to Python 3.12
+
+from ._distutils.ccompiler import CCompiler
+from ._distutils.sysconfig import customize_compiler
+from ._distutils.ccompiler import new_compiler
```

Comment:
is `SETUPTOOLS_USE_DISTUTILS` an env var ? I assume we should adapt behaviour according to its value rather than *always* importing from `distutils`, since the whole point is to be able to do without it. Or am I missing something ?

Reply:
Yes `SETUPTOOLS_USE_DISTUTILS` is an env var.

Setuptools includes a [`MetaPathFinder`](https://github.com/pypa/setuptools/blob/417018d3299bb33e87d2fccec48790cf3e9dd797/_distutils_hack/__init__.py#L88)[^1] that takes into consideration the value of `SETUPTOOLS_USE_DISTUTILS`.

Unless `SETUPTOOLS_USE_DISTUTILS=stdlib` is given, this `MetaPathFinder` will overwrite the importlib machinery to load `distutils` from the `setuptools._distutils` folder.

[^1]: Refs: https://docs.python.org/3/reference/import.html#import-hooks and  https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder

Reply:
oh wow, that rocks. Thanks !